### ISSUE TITLE
Filtering by event type should be configurable

### ISSUE THREAD

**Issue by earthquakesan** [big-data-europe/mu-swarm-logger-service#9]
In [events.py](https://github.com/big-data-europe/mu-swarm-logger-service/blob/master/muswarmlogger/events.py#L66) right now only events of "container" type are getting through. There are 4 general event types "container", "network", "plugin" and "volume". User should be able to configure those.

Imo when deploying with docker containers, the best approach is to give configuration possibility via env vars:
```
EVENT_FILTER = os.environ.get("EVENT_FILTER", "container,plugin").split(",")
```

**Comment by cecton** [comment]
Hello @earthquakesan :) nice to meet you! This is my first contact.

I will split my questions in two parts:

1.

> There are 4 general event types "container", "network", "plugin" and "volume". User should be able to configure those.

Yes, we need to declare the 3 other classes in `events.py`. I would do it only when it starts being used.

2. 

> Imo when deploying with docker containers, the best approach is to give configuration possibility via env vars:

I don't see the point of filtering the Docker events type based on an environment variable. This service is not "dynamic": the loggers (in the modules in the directory "loggers") are the only way to catch the events, it's not meant to be dynamic at this point. But maybe you would like to?

What I mean is that technically speaking the service watch all the events, and the functions registered using `@register_event` filter on what they need already. Adding an environment variable to prevent all the events to arrive will only make the behavior of the application "weird" like: "I have registered my event X, I expect event of type Y but for some reason, I never receive them".

**Comment by earthquakesan** [comment]
Hi @cecton! Nice to meet you too -)

> Yes, we need to declare the 3 other classes in events.py. I would do it only when it starts being used.

Ok, gotcha. Do you have use cases already in mind? I am thinking to use it for monitoring health of Docker Swarm. This will require a CQL-enabled triplestore to register event queries and fire notifications when e.g. some critical containers die/restart.

> "I have registered my event X, I expect event of type Y but for some reason, I never receive them".

I need to read the code more carefully, missed the filters. This is irrelevant then.

**Comment by cecton** [comment]
Well actually you couldn't know, I added the doc about it yesterday because @fr0gs told me it would be useful: https://github.com/big-data-europe/mu-swarm-logger-service/blob/master/docs/INTERNALS.md

I'm a bit too used to be alone on some projects XD

> Ok, gotcha. Do you have use cases already in mind? 

Not yet. I'm not sure I'm suppose to produce the use cases though. Writing all the events to the SPARQL endpoint wouldn't be bad anyway. And I made a code generator so it shouldn't take too much time.
